src/csp.ts

```diff
@@ -0,0 +1,157 @@
+import { HTMLRewriter } from "./html_rewriter_wrapper.ts";
+import { Element } from "./types.d.ts";
+
+import { default as init } from "../pkg/csp_nonce_html_transformer.js";
+await init();
+export type Params = {
+  /**
+   * When true, uses the `Content-Security-Policy-Report-Only` header instead
+   * of the `Content-Security-Policy` header. Setting to `true` is useful for
+   * testing the CSP with real production traffic without actually blocking resources.
+   */
+  reportOnly?: boolean;
+  /**
+   * The relative or absolute URL to report any violations. If left undefined,
+   * violations will not be reported anywhere, which this plugin deploys. If
+   * the response already has a `report-uri` directive defined in its CSP header,
+   * then that value will take precedence.
+   */
+  reportUri?: string;
+  /**
+   * When true, adds `'unsafe-eval'` to the CSP for easier adoption. Set to
+   * `false` to have a safer policy if your code and code dependencies does
+   *  not use `eval()`.
+   */
+  unsafeEval?: boolean;
+  strictDynamic?: boolean;
+  unsafeInline?: boolean;
+  self?: boolean;
+  https?: boolean;
+  http?: boolean;
+  /**
+   * A number from 0 to 1, but 0 to 100 is also supported, along with a trailing %.
+   *
+   * You can ramp up or ramp down the inclusion of the `Content-Security-Policy`
+   * header by setting this to a value between `0` and `1`.
+   *
+   * Any value in between `0` and `1` will include the nonce in randomly distributed traffic.
+   *
+   * For example, a value of `0.25` will put the new directives in the `Content-Security-Policy`
+   * header for 25% of responses. The other 75% of responses will have the new directives
+   * in the `Content-Security-Policy-Report-Only` header.
+   */
+  distribution?: string;
+};
+
+const hexOctets: string[] = [];
+
+for (let i = 0; i <= 255; ++i) {
+  const hexOctet = i.toString(16).padStart(2, "0");
+  hexOctets.push(hexOctet);
+}
+
+function uInt8ArrayToBase64String(input: Uint8Array): string {
+  let res = "";
+
+  for (let i = 0; i < input.length; i++) {
+    res += String.fromCharCode(parseInt(hexOctets[input[i]], 16));
+  }
+
+  return btoa(res);
+}
+
+export function csp(originalResponse: Response, params?: Params) {
```

Comment:
I haven't reviewed this in a lot of detail because I assume it was copied over from the other entry point?

Reply:
That's correct 👍